api/src/main/java/org/openmrs/module/cmrreports/api/reporting/library/Cohorts.java

```diff
@@ -0,0 +1,148 @@
+		patientInYearRange.addParameter(new Parameter("effectiveDate", "effectiveDate", Date.class));
+		patientInYearRange.getSearches().put("1", new Mapped<CohortDefinition>(ageCohort, ParameterizableUtil.createParameterMappings("effectiveDate=${effectiveDate}")));
+		patientInYearRange.getSearches().put("2", new Mapped<CohortDefinition>(genderCohort, null));
+		patientInYearRange.setCompositionString("1 and 2");
+		return patientInYearRange;
+		
+	}
+	
+	public static CompositionCohortDefinition getAllPatientsByGender(GenderCohortDefinition males, GenderCohortDefinition females) {
+		
+		CompositionCohortDefinition allPatients = new CompositionCohortDefinition();
+		allPatients.setName("patientInYearRangeEnrolledInHIVStarted");
+		allPatients.addParameter(new Parameter("effectiveDate", "effectiveDate", Date.class));
+		allPatients.getSearches().put("1", new Mapped<CohortDefinition>(males, null));
+		allPatients.getSearches().put("2", new Mapped<CohortDefinition>(females, null));
+		allPatients.setCompositionString("1 or 2");
+		return allPatients;
+	}
+	
+	public static List<AgeCohortDefinition> getAllAgeRanges() {
+		
+		AgeCohortDefinition PatientBelow1Year = Cohorts.patientWithAgeBelow(1);
+		PatientBelow1Year.setName("PatientBelow1Year");
+		AgeCohortDefinition PatientBetween1And4Years = Cohorts.createXtoYAgeCohort("PatientBetween1And4Years", 1, 4);
+		AgeCohortDefinition PatientBetween5And9Years = Cohorts.createXtoYAgeCohort("PatientBetween5And9Years", 5, 9);
+		AgeCohortDefinition PatientBetween10And14Years = Cohorts.createXtoYAgeCohort("PatientBetween10And14Years", 10, 14);
+		AgeCohortDefinition PatientBetween15And19Years = Cohorts.createXtoYAgeCohort("PatientBetween15And19Years", 15, 19);
+		AgeCohortDefinition PatientBetween20And24Years = Cohorts.createXtoYAgeCohort("PatientBetween20And24Years", 20, 24);
+		AgeCohortDefinition PatientBetween25And49Years = Cohorts.createXtoYAgeCohort("PatientBetween25And49Years", 25, 49);
+		AgeCohortDefinition PatientBetween50YearsAndAbove = Cohorts.patientWithAgeAbove(50);
+		PatientBetween50YearsAndAbove.setName("PatientBetween50YearsAndAbove");
+		
+		ArrayList<AgeCohortDefinition> allAgeRanges = new ArrayList<AgeCohortDefinition>();
+		allAgeRanges.add(PatientBelow1Year);
+		allAgeRanges.add(PatientBetween1And4Years);
+		allAgeRanges.add(PatientBetween5And9Years);
+		allAgeRanges.add(PatientBetween10And14Years);
+		allAgeRanges.add(PatientBetween15And19Years);
+		allAgeRanges.add(PatientBetween20And24Years);
+		allAgeRanges.add(PatientBetween25And49Years);
+		allAgeRanges.add(PatientBetween50YearsAndAbove);
+		return allAgeRanges;
+	}
+	
+	/*
+	 * Just to demo that you can use SQL queries for cohort definitions. You can
+	 * achieve the same here using a Program Enrollment Query and a
+	 * VisitCohortDefinition from the reporting module
+	 */
+	public static SqlCohortDefinition getPatientEnrolledInProgramWithUPECVisit(Program HIVProgram, VisitType UPECVisitType) {
```

Comment:
@rubailly Getting a null pointer here, when there's no patients for that query:

java.lang.NullPointerException
	org.openmrs.module.cmrreports.api.reporting.library.Cohorts.getPatientEnrolledInProgramWithUPECVisit(Cohorts.java:148)
	org.openmrs.module.cmrreports.api.reporting.SetupPECGReport.createReportDefinition(SetupPECGReport.java:70)
	org.openmrs.module.cmrreports.api.reporting.SetupPECGReport.setup(SetupPECGReport.java:44)
	org.openmrs.module.cmrreports.web.controller.CMRReportsController.registerPECG(CMRReportsController.java:33)

Reply:
This was caused by the UPECVisit type not existing.